### PR TITLE
Skip layout dotfiles

### PR DESCRIPTION
**Pull request by durka**
Don't try to read dotfiles in the layout directory. In my case these were vim swap files which don't contain valid UTF-8.

This PR is based on #32, not master, so I'll rebase after (if) that is merged.

### CODE REVIEW

**Comment by durka** [comment]
Rebased.

File: src/cobalt.rs [hunk 1]
```diff
@@ -23,14 +23,16 @@ pub fn build(source: &Path, dest: &Path, layout_str: &str, posts_str: &str) -> i
     // filename -> text content to the layout map
     for entry in walker.filter_map(|e| e.ok()).filter(|e| e.file_type().is_file()) {
```
Comment (johannhof) [inline src/cobalt.rs]: we already apply a filter on the files here, maybe you can create a filter function at the top level that checks if it's a file and if it doesn't start with ".", instead. We could also use that function in other places then.

Comment (durka) [inline src/cobalt.rs]: Yeah, seems like there should be a utility function for creating `WalkDir`s with various filtering options.

**Comment by johannhof** [comment]
@durka are you still working on this? :)

I'd love to get this merged soon, as it's prerequisite to fixing the problems from https://github.com/edunham/cobaltsite. I think we should just ignore dotfiles globally, in all walkdir filter functions.

If I can assist you in any way, let me know.

**Comment by durka** [comment]
Yes. Currently thinking about how best to abstract this out in current Rust where you can't return unboxed iterators. Maybe a macro?

**Comment by durka** [comment]
@johannhof see new commits. I did add a test so not sure about the coveralls error.

**Comment by johannhof** [comment]
Interesting solution with the macro, I would've just defined a function that you can pass into `filter`, but that works for me as well.

Test coverage seems to be plummeting because no additional files are copied over into the target directory. If you'd put a random file into your test it should get copied and then that code will be triggered, it's also nice to test that this actually works. Could you take care of that?

Thanks! :+1:

**Comment by durka** [comment]
Hmm, that didn't fix it...

**Comment by durka** [comment]
Oh, it did fix it, but Github didn't show the new result! Ready to merge, then.

**Comment by johannhof** [comment]
:+1: thanks so much for working on this

**Pull request merged**